content/docs/command-reference/run.md

```diff
@@ -125,29 +282,50 @@ data pipeline (e.g. random numbers, time functions, hardware dependency, etc.)
-  with numeric values or other information that describes a model (or any other
-  regular output). See `dvc metrics` to learn more about using metrics.
-
-- `-M <path>`, `--metrics-no-cache <path>` - the same as `-m` except that files
-  are not tracked by DVC. It means that they are not cached, and it's up to a
-  user to save and version control them. This is typically desirable with metric
-  files, because they are small enough to be tracked by Git directly. See also
-  the difference between `-o` and `-O`.
-
-- `-f <filename>`, `--file <filename>` - specify a path and/or file name for the
-  stage file generated by this command (e.g. `-f stages/stage.dvc`). This
-  overrides the default file name: `<file>.dvc`, where `<file>` is the file name
-  of the first output or metric.
+- `-p [<path>:]<params_list>`, `--params [<path>:]<params_list>` - specify a set
+  of [parameter dependencies](/doc/command-reference/params) the stage depends
+  on, from a parameters file. This is done by sending a coma separated list as
+  argument, e.g. `-p learning_rate,epochs`. The default parameters file name is
+  `params.yaml`, but this can be redefined with a prefix in the argument sent to
+  this option, e.g. `-p parse_params.yaml:threshold`. See `dvc params` to learn
+  more about parameters.
+
+- `-m <path>`, `--metrics <path>` - specify a metrics file produces by this
+  stage. This option behaves like `-o` but registers the file in a `metrics`
+  field inside the `dvc.yaml` stage. Metrics are usually small, human readable
+  files (JSON or YAML) with scalar numbers or other simple information that
+  describes a model (or any other data artifact). See `dvc metrics` to learn
+  more about _metrics_.
+
+- `-M <path>`, `--metrics-no-cache <path>` - the same as `-m` except that DVC
+  does not track the metrics file. This means that the file is not cached, so
+  it's up to the user to save and version control it. This is typically
+  desirable with _metrics_ because they are small enough to be tracked with Git
+  directly. See also the difference between `-o` and `-O`.
+
+- `--plots <path>` - specify a plot metrics file produces by this stage. This
+  option behaves like `-o` but registers the file in a `plots` field inside the
+  `dvc.yaml` stage. Plot metrics are data series stored in tabular (CSV or TSV)
+  or hierarchical (JSON or YAML) files, with complex information that describes
+  a model (or any other data artifact). See `dvc plots` to learn more about
+  plots.
+
+- `--plots-no-cache <path>` - the same as `--plots` except that DVC does not
+  track the plots metrics file. This means that the file is not cached, so it's
+  up to the user to save and version control it. See also the difference between
+  `-o` and `-O`.
+
+- `--external` - allow outputs that are outside of the DVC repository. See
+  [Managing External Data](/doc/user-guide/managing-external-data).
+
+- `--file <path>` - specify name of the YAML file this command will generate or
```

Comment:
Not true. `--file` is ignored for any stages created with `-n`.

Reply:
Yeah that's right but so hmmm what does --file do? This is what the master branch of the core repo outputs:

`  --file <filename>     Specify name of the DVC-file this command will generate.`

We don't even have DVC-files anymore!

Reply:
Also

> ges created with -n

-n is always required no? Lil confused rn

Reply:
Oops, looks like `--single-stage` is hidden. `--single-stage` flag is available in `dvc run` that keeps the same behavior as the 0.X dvc had, regarding single stage files. This `-f` is a vestige of that. We might reuse `-f` if we introduce custom named dvc.yaml file someday. But, for `dvc.yaml`, it's ignored, so, we can just drop it from the documentation altogether.

Reply:
Wait, this is a command reference. So, maybe, we should document `-f` and `--single-stage` here too.

Reply:
Interesting! I didn't know about `--single-stage`.

> for dvc.yaml, it's ignored, so, we can just drop it from the documentation altogether

Yep, I'll just remove it for now. But it should probably be hidden from the help output in core too? Idk 😕

Reply:
@skshetry @efiop let's hide in DVC core if --single-stage is hidden.

Reply:
... and it's ignored for dvc.yaml

Reply:
Created #4084